### ISSUE TITLE
Local to Scene ViewportTextures proxy is invalid.

### ISSUE THREAD

**Issue by STUDIOEMBYR** [godotengine/godot#68371]
### Godot version

4.0 beta 4

### System information

Windows 11, Vulkan, RTX 2070Super

### Issue description

I am porting over a setup that I had working in Godot 3.5 for rendering the world at a lower resolution and then upscaling without pixel jitter that involves using two viewport textures. This works in the editor but when the game launches I get the following errors and I am met with a black screen.

![Screenshot_35](https://user-images.githubusercontent.com/20043270/200315873-2b3b8374-ba26-40f7-9fc6-e79270a46615.png)

### Steps to reproduce

![Screenshot_32](https://user-images.githubusercontent.com/20043270/200315252-3f21344d-c94c-4ea4-8e75-2ecc4462ebf8.png)

![Screenshot_33](https://user-images.githubusercontent.com/20043270/200315268-44e2b215-1cc3-4108-9ba8-efaa767640d4.png)

When the game is launched you will be met with a black screen.

![Screenshot_34](https://user-images.githubusercontent.com/20043270/200315279-351a68b2-c5d7-468f-b568-54c217ee4388.png)

### Minimal reproduction project

[project.zip](https://github.com/godotengine/godot/files/9951779/project.zip)

**Comment by Calinou** [comment]
Note that rendering 2D and 3D at different scale factors is supported out of the box in 4.0.beta (**Scaling 3D** project settings), but nearest-neighbor filtering isn't implemented yet for 3D scaling.

**Comment by STUDIOEMBYR** [comment]
> Note that rendering 2D and 3D at different scale factors is supported out of the box in 4.0.beta (**Scaling 3D** project settings), but [nearest-neighbor filtering isn't implemented yet for 3D scaling](https://github.com/godotengine/godot-proposals/issues/4697).

Yeah I'm very specifically rendering at a low resolution, blitting it to a buffet 4x it's size to maintain pixelation and then stretching that texture using bilinear filtering so that I avoid issues with different screen sizes that aren't multiples of the native res having inconsistently scaled pixels.

**Comment by STUDIOEMBYR** [comment]
Reproduced on macOS 13, M1 Max.

**Comment by STUDIOEMBYR** [comment]
When running in gl_compatibility mode the viewport does render but incorrectly, the cube color can actually be different between runs.

![image](https://user-images.githubusercontent.com/20043270/200414112-cfa8f5a2-9b77-448c-9729-bcbc4bd1cba2.png)

**Comment by STUDIOEMBYR** [comment]
When running in gl_compatibility mode the viewport does render but incorrectly. The color of the cube is not deterministic but I get cyan most often. I have had it render the cube correctly only once however the world environment was not respected.

![image](https://user-images.githubusercontent.com/20043270/200414431-8386978c-36be-4846-8fb3-cbb7da66a94e.png)

**Comment by Rindbee** [comment]
Error messages could be fixed by #64388.

The rest of the problem can be solved by adding the environment and sun.

https://user-images.githubusercontent.com/30386067/200495932-c64ebdf5-9f3a-49bb-8790-ad336feba7ca.mp4

**Comment by STUDIOEMBYR** [comment]
> Error messages could be fixed by #64388.
> 
> The rest of the problem can be solved by adding the environment and sun.
> 0.mp4

Ahhh derp! I assumed the default world env would pass through but that's my bad. Obviously the error messages are still a problem but this unblocks me.

**Comment by eh-jogos** [comment]
Error message still happening in 4.0rc2, was going to report this today but just found this older issue.

**Comment by Gegbee** [comment]
Can confirm that this error is happening in Godot 4.0 stable. Though it does not interfere in in editor play mode, exporting the project does not work because the viewports do not show up.

**Comment by Rindbee** [comment]
The error message was fixed by #75751.

**Comment by clayjohn** [comment]
Can anyone reproduce this in 4.1 beta 1?
